apps/api/src/server/repositories/folder.repository.js

```diff
@@ -69,39 +71,274 @@ export const getFolderPath = async (caseId, currentFolderId) => {
 };
 
 /**
+ * adds the caseId to a folder ready for db creation, and recursively adds to all child folders
  *
  * @param {number} caseId
- * @returns {Promise<import('@pins/api').Schema.BatchPayload>}
+ * @param {FolderTemplate} folder
+ * @returns {FolderTemplate}
  */
-export const createFolders = (caseId) => {
-	const createdFolders = {
-		data: defaultCaseFolders.map((folder) => ({ ...folder, caseId }))
-	};
+const mapFolderTemplateWithCaseId = (caseId, folder) => {
+	folder.caseId = caseId;
 
-	return databaseConnector.folder.createMany(createdFolders);
+	if (folder.childFolders && folder.childFolders.create) {
+		for (let /** @type {FolderTemplate} */ child of folder.childFolders.create) {
+			child = mapFolderTemplateWithCaseId(caseId, child);
+		}
+	}
+
+	return folder;
 };
 
 /**
+ * Creates the top level folders on a case using the folder template
+ * and recursively creates all sub folders.
+ * Returns an array of promises
  *
- * @param {number} folderId
- * @returns {Promise<import('@pins/api').Schema.Folder | null>}
+ * @param {number} caseId
+ * @returns {Promise<import('@pins/api').Schema.BatchPayload>[]}
```

Comment:
I did not realise that the response of a `folder.create` function is a `Promise<BatchPayload>`? I thought it was `Promise<Folder>`?

Reply:
I don't know Mayya.  It was returning a BatchPayLoad before, so I did not change it, I just made it an array [].  The compare above looks a bit confusing, I think it is showing the params for a separate fn.  I looked back through code before this ticket and it was BatchPayLoad.

Reply:
I think `BatchPayload` was accurate as of using `createMany`, but not any longer as we are using `create`

Reply:
ok, I will change to Promise<Folder>[] and re-test